lib/async_hooks.js

```diff
@@ -223,15 +224,20 @@ class AsyncResource {
     return this[trigger_async_id_symbol];
   }
 
-  bind(fn, thisArg = this) {
+  bind(fn, thisArg) {
     validateFunction(fn, 'fn');
-    const ret =
-      FunctionPrototypeBind(
-        this.runInAsyncScope,
-        this,
-        fn,
-        thisArg);
-    ObjectDefineProperties(ret, {
+    const runInAsyncScope = this.runInAsyncScope;
```

Comment:
Any reason not to use destructuring here? If you don't like the syntax, we could also ditch the variable altogether and access `resource.runInAsyncScope` directly.
Also, please put those variable definition inside the if, so the engine can skip the assignment when `thisArg` is defined.

Reply:
> Any reason not to use destructuring here?

I was using destructuring before, this must be a leftover of trying out stuff, I can change it.

> Also, please put those variable definition inside the if block, so the engine can skip the assignment when thisArg is defined.

Isn't the whole point of using `const` that the engine is able to avoid the assignment when it's not needed?

Reply:
> Isn't the whole point of using `const` that the engine is able to avoid the assignment when it's not needed?

No, `const` means the variable will always reference the same value/object. The engine must still allocate some memory to keep track of the new identifier for this scope.